src/tests/multihost/ipa/test_subid_ranges.py

```diff
@@ -0,0 +1,132 @@
+
+
+@pytest.mark.usefixtures('install_list_subid_ranges',
+                         'setup_ipa_client')
+@pytest.mark.tier1
+class TestSubid(object):
+    """
+    This is for ipa bugs automation
+    """
+    def test_subid_feature(self, multihost,
+                           create_bkp_for_subid_files):
+        """
+        :Title: support subid ranges managed by FreeIPA
+        :id: 50bcdc28-00c8-11ec-bef4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1803943
+        :steps:
+            1. Generate subid for user admin
+            2. Test newuidmap command
+            3. Test newgidmap command
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        (result, result1, exit_status) = ssh1.execute_cmd('kinit',
+                                                          stdin=test_password)
+        assert exit_status == 0
+        (result, result1, exit_status) = ssh1.exec_command('klist')
+        assert user in str(result1.read())
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                           f"subid-generate "
+                                                           f"--owner={user}")
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                           f"subid-find "
+                                                           f"--owner "
+                                                           f"{user}")
+        user_details = result1.readlines()
+        uid_start = int(user_details[5].split(': ')[1].split('\n')[0])
+        uid_range = int(user_details[6].split(': ')[1].split('\n')[0])
+        gid_start = int(user_details[7].split(': ')[1].split('\n')[0])
+        gid_range = int(user_details[8].split(': ')[1].split('\n')[0])
+        ssh2 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        cmd = execute_cmd(multihost, "ps -ef | grep bash").stdout_text
+        find_admin = f"ps -ef | grep bash | grep {user}"
+        if user in cmd:
+            proces_id = [int(i)
+                         for i in execute_cmd(multihost,
+                                              find_admin).stdout_text.split()
+                         if i.isdigit()][0]
+            multihost.client[0].run_command(f'kill -9 {proces_id}')
+        (results1, results2, results3) = ssh1.exec_command('unshare -U bash')
+        proces_id = [int(i)
+                     for i in execute_cmd(multihost,
+                                          find_admin).stdout_text.split()
+                     if i.isdigit()][0]
+        ssh1.exec_command(f"newuidmap {proces_id} "
+                          f"{uid_start} {uid_start + 1} 1")
```

Comment:
Please add a check for error code returned by `new?idmap`

Reply:
That is not necessary , script will fail at this point if something goes wrong . And we need this.

Reply:
Good test (and code in general) validates every step where it makes sense.
Please check return code and fail the test in case of any error.

Reply:
This is not needed as these lies are already there to validate the results .

    result = execute_cmd(multihost, f"cat /proc/{proces_id}/uid_map")
    assert f'{uid_start} {uid_start + 1}' in result.stdout_text

Reply:
Checking return code of `new?idmap` gives more fine grained validation (i.e. allow to understand fail reason better).